### ISSUE TITLE
[Feature Request] Add --profile flag back.

### ISSUE THREAD

**Issue by LackoDan** [aws-quickstart/taskcat#416]
I have credentials and config for various profiles stored in `~/.aws`, but I intentionally do not use a default profile. Instead, I prefer to pass a specific profile name to the AWS CLI using the `--profile` argument. This has worked very well for all AWS tools I've used so far.

The README states that using shared credentials or config files should be possible, and everything does indeed work if I set the `AWS_PROFILE` environment variable to the profile name I want to use. 

However, is there an option to pass a profile name to taskcat instead of using an environment variable? [This article](https://aws.amazon.com/blogs/infrastructure-and-automation/a-deep-dive-into-testing-with-taskcat/) shows how to do just that using the `-P` argument, but that no longer seems to be supported. 

Could anyone please point me in the right direction?

**Comment by andrew-glenn** [comment]
@LackoDan 

We'll consider this a feature request to re-add the profile flag. We made an intentional design decision to not mess with credentials on the command line, but I think the profile is an exception. 

The project config (.taskcat.yml) and global config (~/.taskcat.yml) files support an 'auth' directive.  Here's an example of using a specific profile globally and specific to the project. 

Project Config

```
project:
  owner: quickstart-eng@amazon.com
  name: complex-auth-config-file
  regions:
    - ap-northeast-1
    - ap-northeast-2
    - ap-south-1
    - ap-southeast-1
    - ap-southeast-2
    - eu-central-1
    - eu-west-1
    - eu-west-2
    - us-east-1
    - us-east-2
    - us-west-2
    - sa-east-1
    - me-south-1
    - ap-east-1
  parameters: {
    "AvailabilityZones": "$[taskcat_genaz_3]",
    "ByteValue": 1,
    "GlobalOverrideTest": "override",
    "LicenseContentTest": "$[taskcat_getlicensecontent]/aws-quickstart-datasets/taskcat-license-param-test/example.txt",
    "LocalOverrideTest": "override",
    "PasswordA": "$[taskcat_genpass_8A]",
    "PasswordAConfirm": "$[taskcat_getval_PasswordA]",
    "PasswordB": "$[taskcat_genpass_32S]",
    "RandomNumber": "$[taskcat_random-numbers]",
    "RandomString": "$[taskcat_random-string]",
    "UUID": "$[taskcat_genuuid]"
  }
  auth:
    default: profile-used-for-every-test-in-this-project
    ap-east-1: hongkong
tests:
  json-test:
    template: templates/debug.template
    auth:
      default: special-use-case-profile-name-for-this-test
    parameters:
      StackName: TestStack
```


Project level config

```
general:
  auth:
    default: profile-for-everything
```

I also threw in an example around region-specific credentials. The `auth` section accepts either `default` or `<region_name>`. 

Until we can add in the profile flag, does this get you in the right direction?

**Comment by LackoDan** [comment]
Hey @andrew-glenn, thanks for the quick and clear response. Very helpful! This will be perfect for now. 

My question is fully answered, so I would normally close the issue, but perhaps you would like to keep it open as a feature request?

**Comment by andrew-glenn** [comment]
Glad to hear it. Yep, I'll leave it open until we can circle back

**Comment by andrew-glenn** [comment]
@LackoDan - Circling back on this, I think any value for the --profile flag would only apply to the `default` profile. Any region-specific stuff should be defined in the config files.

Ex:
`--profile foobar`

Would be equal to:

Global config
```
general:
  auth:
    default: foobar
```

or

Project Config
```
project:
  auth:
    default: foobar
```

and would supercede any values there due to inheritence.

Adding
```
--profile us-east-1=foobar,us-east-2=asklfakfj
```

Adds a lot of complexity, and a use-case like that really should be written down somewhere. 

All of that to ask... will that fit your use case? I want to make sure I don't back you into a corner unintentionally.

**Comment by andrew-glenn** [comment]
_This feature will be included in the `0.9.18` release of taskcat. In the interim, a development release is available._

`pip install taskcat==0.9.18.dev7`
